app/src/marketplace/containers/ProductController/ValidationContextProvider.jsx

```diff
@@ -154,11 +154,19 @@ function useValidationContext(): ContextProps {
             clearStatus('pricePerSecond')
         }
 
-        // Set pending fields
+        // Set pending fields, a change is marked pending if there was a saved pending change or
+        // we made a change that is different from the loaded product
         const changes = getPendingChanges(product)
         const isPublic = isPublished(product)
         PENDING_CHANGE_FIELDS.forEach((field) => {
-            setPendingChange(field, (field in changes) || (isPublic && isTouched(field) && !isEqual(product[field], originalProduct[field])))
+            let fieldsAreEqual = isEqual(product[field], originalProduct[field])
+
+            // special case, imageUrl will be updated if there is a new image uploaded
```

Comment:
I wonder if we could invalidate the old `imageUrl` while we set `newImageToUpload: image` in [`useEditableProductActions#L67`](https://github.com/streamr-dev/streamr-platform/blob/PLATFORM-1332-republish-cover-image/app/src/marketplace/containers/ProductController/useEditableProductActions.js#L67). This could potentially eliminate the need for a special case. I'm not sure how far it'd fly though…

Thoughts?

Reply:
Thanks, that seems to be neater solution 🎉 Fixed in https://github.com/streamr-dev/streamr-platform/pull/959/commits/9ecc2c9740f5f0a429dba017b13ea9162c9ac7ff